src/js/elements/menu.js

```diff
@@ -1,174 +1,194 @@
-
-        if (groups !== null && typeof groups.subindicators !== 'undefined') {
-            groups.subindicators.forEach((subindicator) => {
-                const newSubIndicatorElement = indicatorItemTemplate.cloneNode(true);
-                $(".truncate", newSubIndicatorElement).text(subindicator);
-                $(newSubIndicatorElement).attr('title', subindicator);
-
-                wrapper.append(newSubIndicatorElement);
-
-                const parents = {
-                    category: category,
-                    subcategory: subcategory,
-                    indicator: indicator
-                }
-
-                $(newSubIndicatorElement).on("click", (el) => {
-                    setActive(el);
-                    if (subindicatorCallback != undefined)
-                        subindicatorCallback({
-                            el: el,
-                            data: data,
-                            indicatorTitle: indicator,
-                            selectedSubindicator: subindicator,
-                            indicators: indicators,
-                            parents: parents,
-                            choropleth_method: choropleth_method,
-                            indicatorId: indicatorId
-                        })
-                });
-            });
+export class DataMapperMenu extends Component {
+  constructor(parent) {
+    super(parent);
+    this._data = null;
+    this._subindicatorCallback = null;
+  }
+
+  init(data, subindicatorCallback) {
+    this._data = data;
+    this._subindicatorCallback = subindicatorCallback;
+    this._model = new LoadMenuModel();
+    this.main();
+  }
+
+  addSubIndicators(wrapper, category, subcategory, indicator, groups, indicators, choropleth_method, indicatorId) {
+    $(".data-category__h3", wrapper).remove();
+    $(".data-category__h4", wrapper).remove();
+
+    if (groups !== null && typeof groups.subindicators !== 'undefined') {
+      groups.subindicators.forEach((subindicator) => {
+        const newSubIndicatorElement = indicatorItemTemplate.cloneNode(true);
+        $(".truncate", newSubIndicatorElement).text(subindicator);
+        $(newSubIndicatorElement).attr('title', subindicator);
+
+        wrapper.append(newSubIndicatorElement);
+
+        const parents = {
+          category: category,
+          subcategory: subcategory,
+          indicator: indicator
         }
+
+        $(newSubIndicatorElement).on("click", (el) => {
+          this.setActive(el);
+          if (this._subindicatorCallback != undefined)
+            this._subindicatorCallback({
+              el: el,
+              data: this._data,
+              indicatorTitle: indicator,
+              selectedSubindicator: subindicator,
+              indicators: indicators,
+              parents: parents,
+              choropleth_method: choropleth_method,
+              indicatorId: indicatorId
+            })
+        });
+      });
```

Comment:
Not for this PR, but we should replace this with an event - the callback isn't nice

Reply:
Noted!